akka-docs/rst/java/code/docs/http/javadsl/server/directives/BasicDirectivesExamplesTest.java

```diff
@@ -0,0 +1,788 @@
+    );
+
+    final Route route = mapResponseHeaders(headers -> {
+      headers.removeIf(header -> header.lowercaseName().equals("id"));
+      return headers;
+    }, () -> echoRequestHeaders);
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/").addHeaders(
+      Arrays.asList(RawHeader.create("id", "12345"),RawHeader.create("id2", "67890"))))
+      .assertHeaderKindNotExists("id")
+      .assertHeaderExists("id2", "67890");
+    //#mapResponseHeaders
+  }
+
+  @Ignore("Not implemented yet")
+  @Test
+  public void testMapInnerRoute() {
+    //#mapInnerRoute
+    // TODO: implement mapInnerRoute
+    //#mapInnerRoute
+  }
+
+  @Test
+  public void testMapRejections() {
+    //#mapRejections
+    // ignore any rejections and replace them by AuthorizationFailedRejection
+    final Route route = mapRejections(
+      rejections -> Collections.singletonList((Rejection) Rejections.authorizationFailed()),
+      () -> path("abc", () -> complete("abc"))
+    );
+
+    // tests:
+    runRouteUnSealed(route, HttpRequest.GET("/"))
+      .assertRejections(Rejections.authorizationFailed());
+    testRoute(route).run(HttpRequest.GET("/abc"))
+      .assertStatusCode(StatusCodes.OK);
+    //#mapRejections
+  }
+
+  private boolean existsAuthenticationFailedRejection(Iterable<Rejection> rejections) {
+    for (Rejection r : rejections) {
+      if (r instanceof akka.http.scaladsl.server.AuthenticationFailedRejection) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private Optional<Object> neverAuth(Optional<ProvidedCredentials> creds) {
+    return Optional.empty();
+  }
+
+  private Optional<Object> alwaysAuth(Optional<ProvidedCredentials> creds) {
+    return Optional.of("id");
+  }
+
+  @Test
+  public void testRecoverRejections() {
+    //#recoverRejections
+    final Route originalRoute = pathPrefix("auth", () ->
+      route(
+        path("never", () ->
+          authenticateBasic("my-realm", this::neverAuth, obj ->  complete("Welcome to the bat-cave!"))
+        ),
+        path("always", () ->
+          authenticateBasic("my-realm", this::alwaysAuth, obj -> complete("Welcome to the secret place!"))
+        )
+      )
+    );
+
+    final Route route = recoverRejections(rejections -> {
+      if (existsAuthenticationFailedRejection(rejections)) {
+        return new akka.http.scaladsl.server.RouteResult.Complete(
+          (akka.http.scaladsl.model.HttpResponse)
+            HttpResponse.create().withEntity("Nothing to see here, move along."));
+      } else if (!rejections.iterator().hasNext()) { // see "Empty Rejections" for more details
+        return new akka.http.scaladsl.server.RouteResult.Complete(
+          (akka.http.scaladsl.model.HttpResponse)
+            HttpResponse.create().withStatus(StatusCodes.NOT_FOUND).withEntity("Literally nothing to see here."));
+      } else {
+        return new akka.http.scaladsl.server.RouteResult.Rejected(Util.convertIterable(rejections));
```

Comment:
Besides, I alse refactor the `mapSettings/withSettings/extractSettings` to use `akka.http.javadsl.settings.RoutingSettings` instead of `akka.http.scaladsl.settings.RoutingSettings`